presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -296,12 +297,18 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
             builder.add(new AddExchanges(metadata, sqlParser));
         }
 
-        builder.add(new PickLayout(metadata));
+        builder.add(
+                new IterativeOptimizer(
+                        stats,
+                        ImmutableList.of(new com.facebook.presto.sql.planner.optimizations.PickLayout(metadata)),
+                        ImmutableSet.of(
+                                new PickTableLayoutForPredicate(metadata),
+                                new PickTableLayoutWithoutPredicate(metadata))));
```

Comment:
You might want to consider to use a `RuleSet` to group this to rules together, that way you could merge tests as well